src/test/java/com/fryrank/dal/ReviewDALTests.java

```diff
@@ -24,21 +31,44 @@ public class ReviewDALTests {
     @InjectMocks
     ReviewDALImpl reviewDAL;
 
+    private static final String TEST_RESTAURANT_ID = "1";
+
     @Test
-    public void testGetAllReviewsByRestaurantId() throws Exception {
-        Query query = new Query();
-        query.addCriteria(where("restaurantId").is("1"));
-        List<Review> expectedReviews = new ArrayList<>() {
+    public void testGetAllReviewsByRestaurantId_happyPath() throws Exception {
+        final Query query = new Query();
+        query.addCriteria(where(RESTAURANT_ID_KEY).is(TEST_RESTAURANT_ID));
+
+        final List<Review> expectedReviews = new ArrayList<>() {
             {
                 add(new Review("review_id_1", "restaurant_id_1", "author_id_1", 5.0 , "title_1", "body_1"));
                 add(new Review("review_id_2", "restaurant_id_2", "author_id_2", 7.0 , "title_2", "body_2"));
             }
         };
         when(mongoTemplate.find(query, Review.class)).thenReturn(expectedReviews);
 
-        List<Review> actualReviews = reviewDAL.getAllReviewsByRestaurantId("1");
+        final AggregationResults<RestaurantAvgScore> aggregationResults = new AggregationResults<>(List.of(
+                new RestaurantAvgScore(RESTAURANT_ID_KEY, 6f)), new Document());
+        when(mongoTemplate.aggregate(any(Aggregation.class), anyString(), eq(RestaurantAvgScore.class))).thenReturn(aggregationResults);
+
+        final GetAllReviewsOutput expectedOutput = new GetAllReviewsOutput(expectedReviews, 6f);
+        final GetAllReviewsOutput actualOutput = reviewDAL.getAllReviewsByRestaurantId(TEST_RESTAURANT_ID);
+        assertEquals(actualOutput, expectedOutput);
+    }
+
+    @Test
+    public void testGetAllReviewsByRestaurantId_noReviews() throws Exception {
+        final Query query = new Query();
+        query.addCriteria(where(RESTAURANT_ID_KEY).is(TEST_RESTAURANT_ID));
+
+        final List<Review> expectedReviews = List.of();
+        when(mongoTemplate.find(query, Review.class)).thenReturn(expectedReviews);
+
+        final AggregationResults<RestaurantAvgScore> aggregationResults = new AggregationResults<>(List.of(), new Document());
+        when(mongoTemplate.aggregate(any(Aggregation.class), anyString(), eq(RestaurantAvgScore.class))).thenReturn(aggregationResults);
 
-        assertEquals(actualReviews, expectedReviews);
+        final GetAllReviewsOutput expectedOutput = new GetAllReviewsOutput(expectedReviews, null);
+        final GetAllReviewsOutput actualOutput = reviewDAL.getAllReviewsByRestaurantId(TEST_RESTAURANT_ID);
+        assertEquals(actualOutput, expectedOutput);
     }
```

Comment:
Test for `getAllReviewsByRestaurantId()` with null restaurant ID

Reply:
Sure, adding it. Expected result would be NullPointerException